ES6 refactoring

Schultzer opened the pull request.

Schultzer committed:
Removed curly braces

Schultzer committed:
Refactored filelist

Arrow functions are now everwhere.
For-loops now use `let` instead for `var`.
Export moved to the bottom.

Schultzer committed:
Refactored select

Arrow functions are now everwhere.
For-loops now use `let` instead for `var`.
Export moved to the bottom.

Schultzer committed:
Refactored util

Arrow functions are now everwhere.
For-loops now use `let` instead for `var`.
Export moved to the bottom.

Schultzer committed:
Refactored app

Arrow functions are now everwhere.
Import all modules.
Removed unnecessary comments.

Schultzer committed:
Refactored edit

Arrow functions are now everwhere.
Import all modules.

Schultzer committed:
Refactored report

Arrow functions are now everwhere.
Import all modules.

Schultzer committed:
Refactored contextMenu

Arrow functions are now everwhere.
Export moved to the bottom.
Moved contextMenu to menu folder.

Schultzer committed:
Refactored external_link

Arrow functions are now everwhere.
Export moved to the bottom.
Moved external_link to src/helpers folder.

Schultzer committed:
Refactored crash_reporter

Arrow functions are now everwhere.
Export moved to the bottom.
Moved external_link to src/helpers folder.

Schultzer committed:
Refactored app_menu

Export moved to the bottom.

Schultzer committed:
Refactored dev_menu

Arrow functions are now everwhere.
Export moved to the bottom.

Schultzer committed:
Refactored edit_menu

Export moved to the bottom.

Schultzer committed:
Refactored help_menu

Arrow functions are now everwhere.
Export moved to the bottom.

Schultzer committed:
Refactored window_menu

Export moved to the bottom.

Schultzer committed:
Refactored createWindow()

Arrow functions are now everwhere.
Export moved to the bottom.

Schultzer committed:
Refactored background

Arrow functions are now everwhere.
Import all modules.
Fixed bug with blur().
Removed unnecessary comments.

Schultzer committed:
Removed unnecessary comments.
Changed var to const.

Schultzer committed:
Fixed indention

Schultzer committed:
Added test for removal of file in filelist.

danschultzer commented on .travis.yml:
```diff
@@ -9,6 +9,7 @@ env:
   - ISTANBUL_REPORT_DIR=coverage ISTANBUL_REPORTERS=lcovonly CASHER_TIME_OUT=54000
 
 cache:
+  timeout: 5000
```
Seems excessive to go from 180 to 5000, why 5000?

danschultzer commented on e2e/app.e2e.js:
```diff
@@ -41,6 +41,22 @@ describe('application launch', function () {
     })
 
+    describe('when removing', function () {
+      it('should remove file', function () {
```
Using `should` is not really good specs, it's redundant. http://betterspecs.org/#should

danschultzer replied:
The explanation is incorrect too, should be more like `removes from list`.

danschultzer commented on e2e/app.e2e.js:
```diff
@@ -41,6 +41,22 @@ describe('application launch', function () {
         })
     })
 
+    describe('when removing', function () {
```
Why not arrow method here if this is a full change to ES6?

danschultzer commented on e2e/app.e2e.js:
```diff
@@ -41,6 +41,22 @@ describe('application launch', function () {
+            document.dispatchEvent(new window.Event('remove'))
+          })
+          .isVisible('.table .body .row:nth-child(1)').then(function (visible) {
+            assert.equal(visible, false)
```
This is a weird assertion, you check if the first row is visible, then assert it's not??

danschultzer commented on e2e/mocks.js:
```diff
@@ -1,5 +1,3 @@
-electron.dialog.showOpenDialog = (opts, cb) => {
-  cb([path.join(__dirname, 'support', 'readable.pdf')])
-}
+electron.dialog.showOpenDialog = (opts, cb) => cb([path.join(__dirname, 'support', 'readable.pdf')])
```
Why is this in one line? Two things happen in the logic, so it would be better to keep it on two lines.

danschultzer commented on src/app.js:
```diff
@@ -1,15 +1,14 @@
 import Vue from 'vue'
 import env from './env'
 import { FileList } from './file_list/file_list'
+import crashReporter from './helpers/crash_reporter'
```
Why this? Shouldn't that be in background?

danschultzer commented on src/app.js:
```diff
@@ -71,8 +67,8 @@ document.addEventListener('DOMContentLoaded', function () {
-      successRateLabel: function () {
-        var total = this.result.done.successful / (this.result.done.total || 1) * 100
+      successRateLabel: event => {
+        var total = event.result.done.successful / (event.result.done.total || 1) * 100
```
The `result` key is in the event object?

Schultzer committed:
Clean up

Schultzer committed:
Bug fix

Dont’ allow handleCmdOrCtrlA() or handleCmdOrCtrlBackspace() when there is no filelist view

Schultzer committed:
Fixed arrow functions and for loops

Schultzer committed:
Refactored

danschultzer commented on src/env.js:
```diff
@@ -1,8 +1,5 @@
 
-// The variables have been written to `env.json` by the build process.
-var env = jetpack.cwd(__dirname).read('env.json', 'json')
+const env = jetpack.cwd(__dirname).read('env.json', 'json')
```
I don't think it's a good idea to make this a constant. The environment may be changed on runtime.

danschultzer commented on src/edit.js:
```diff
@@ -154,6 +152,4 @@ function save () {
-function close () {
-  ipcRenderer.send('close-edit')
-}
+function close () { ipcRenderer.send('close-edit') }
```
This should probably be on two lines.

danschultzer commented on src/menu/context_menu.js:
```diff
@@ -1,36 +1,28 @@
-  var isAnyTextSelected = function () {
-    return window.getSelection().toString() !== ''
-  }
+  function isAnyTextSelected () { return window.getSelection().toString() !== '' }
```
This could be on two lines.

danschultzer reviewed (changes requested):
There's a few places where you've changed a method to be a one-liner, but it could've been on several lines to make it more readable.

Schultzer committed:
Clean up

danschultzer commented on src/helpers/crash_reporter.js:
```diff
@@ -1,4 +1,4 @@
-module.exports = function (env) {
+function crashReport (env) {
   if (env.crashSubmitURL) {
     const { crashReporter } = require('electron')
```
Shouldn't this be ES6 `import` `form` too?

danschultzer commented:
Apart from comments (import/from and multiline vs single line) this can be merged in. Might do a quick test of drag n' drop to ensure it works as expected (we don't test for that).
